### PR TITLE
Security chat

### PR DESCRIPTION
**Pull request by maxLogvynyuk**
Fix differents with master. Add axios, socket, integrate with server, create component Avatar messageStatus

### CODE REVIEW

File: src/reducers/chat.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    case GET_ALL_MESSAGES:
+      return {
+        ...state,
+        messages: action.items,
+        isTyping: false,
+      };
+    case CHAT_ADD_MESSAGE:
+      return {
+        ...state,
+        newMessage: action.message,
```
Comment (undead404) [inline src/reducers/chat.js]: So, new message doesn't appear in `messages` array? Is that what expected?

File: src/routes/chat/Chat.js [hunk 2]
```diff
@@ -1,37 +1,84 @@
-import React, { memo, useState, useCallback } from 'react';
+import React, { memo, useState, useCallback, useEffect } from 'react';
 import get from 'lodash/get';
-import PropTypes from 'prop-types';
+import property from 'lodash/property';
+// import PropTypes from 'prop-types';
```
Comment (undead404) [inline src/routes/chat/Chat.js]: Why don't you define propTypes for `Chat` component?

File: src/routes/chat/Chat.js [hunk 3]
```diff
@@ -1,37 +1,84 @@
...
         <div className={s.header}>Security Chat</div>
-        <div className={s.messages}>Messages</div>
-        <input
-          className={s.input}
-          onChange={handleChange}
-          type="text"
-          value={userInput}
+        <div className={s.messagesContent}>
+          <div className={s.messages}>
+            {!messages || null || undefined ? (
```
Comment (undead404) [inline src/routes/chat/Chat.js]: This is redundant, you can simply use `isEmpty(messages)` using lodash's `isEmpty`

File: src/routes/chat/components/avatar/Avatar.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        src={user.avatar}
+      />
+    );
+  }
+  const firstChar = toUpper(`${user.fullname[0]}`);
+  return <div className={s.symbol}>{firstChar}</div>;
+}
+
+Avatar.defaultProps = {
+  user: {},
```
Comment (undead404) [inline src/routes/chat/components/avatar/Avatar.js]: So, it will show big 'U' on empty user?

File: src/routes/chat/components/message/Message.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              className={classNames(`${s.bubble}`, {
+                [s.bubbleIsMe]: isMe,
+              })}
+            >
+              <p>{text}</p>
+            </div>
+            <IconMessageStatus isMe={isMe} isRead={testIsRead} />
+          </div>
+          <span
+            className={classNames(`${s.date}`, {
```
Comment (undead404) [inline src/routes/chat/components/message/Message.js]: Is there any reason to wrap dynamic classes with `${}`?

File: src/routes/chat/components/message/Message.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  useStyles(s);
+  return (
+    <div
+      className={classNames('message', {
+        [s.isMe]: isMe,
+      })}
+    >
+      <div className={s.message}>
+        <span
+          className={classNames('', {
```
Comment (undead404) [inline src/routes/chat/components/message/Message.js]: Static empty string is redundant here

Comment (maxLogvynyuk) [reply src/routes/chat/components/message/Message.js]: Change to "className={classNames( {"   or "className={classNames(null, {" ?

File: src/routes/chat/components/avatar/Avatar.js (hunk 4 above)
Comment (maxLogvynyuk) [reply src/routes/chat/components/avatar/Avatar.js]: Yes in a colored circle

File: src/routes/chat/components/message/Message.js (hunk 5 above)
Comment (maxLogvynyuk) [reply src/routes/chat/components/message/Message.js]: Agree, I change it.

File: src/routes/chat/Chat.js (hunk 3 above)
Comment (maxLogvynyuk) [reply src/routes/chat/Chat.js]: +

File: src/routes/chat/Chat.js (hunk 2 above)
Comment (maxLogvynyuk) [reply src/routes/chat/Chat.js]: When I applied it to messages Eslint show me error "That propTypes defined but never used". The array checked in Messages component. For what element 
useful to applied prop types?

File: src/routes/chat/components/message/Message.js (hunk 6 above)
Comment (undead404) [reply src/routes/chat/components/message/Message.js]: First option, please. `classNames` ignores any falsy values.

File: src/routes/chat/components/message/Message.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        </div>
+      </div>
+    </div>
+  );
+}
+
+Message.defaultProps = {
+  createdAt: null,
+  isMe: false,
+  text: PropTypes.string,
```
Comment (undead404) [inline src/routes/chat/components/message/Message.js]: All default props should be simple values

**Comment by undead404** [comment]
To fix the conflict, remove the lockfile and generate it again

**Comment by maxLogvynyuk** [comment]
Update yarn.lock

File: src/routes/chat/Chat.js [hunk 8]
```diff
@@ -1,37 +1,85 @@
...
         />
       </div>
     </div>
   );
 }
 Chat.whyDidYouRender = true;
-Chat.prototype = {
-  name: PropTypes.string,
-};
 Chat.defaultProps = {
```
[the remark this answers is not in the document]
Comment (undead404) [reply src/routes/chat/Chat.js]: Remove this, it defines default props which doesn't exist

File: src/routes/chat/components/message/Message.js (hunk 5 above)
Comment (undead404) [reply src/routes/chat/components/message/Message.js]: ok, do please

**Comment by maxLogvynyuk** [comment]
Add default state = { messages: [] }. When array of messages.lenght = 0 without default state = { messages: [] }.   case CHAT_ADD_MESSAGE:  is crashing

**Pull request merged**